src/lvm.cpp

```diff
@@ -0,0 +1,451 @@
+                std::string vgName, vgSizeStr, vgFreeStr;
+                if (!(iss >> vgName >> vgSizeStr >> vgFreeStr)) {
+                    LOG_DEBUG("LVM: Failed to split line into expected fields. "
+                              "Line: {}",
+                        cleanLine);
+                    throw std::runtime_error(
+                        "LVM ERROR: Failed to parse volume group information.");
+                }
+
+                // Remove 'G' from size and free space strings
+                vgSizeStr.erase(
+                    std::remove(vgSizeStr.begin(), vgSizeStr.end(), 'G'),
+                    vgSizeStr.end());
+                vgFreeStr.erase(
+                    std::remove(vgFreeStr.begin(), vgFreeStr.end(), 'G'),
+                    vgFreeStr.end());
+
+                // Convert vg_size and vg_free from string to double
+                double vgSizeGB = std::stod(vgSizeStr);
+                double vgFreeGB = std::stod(vgFreeStr);
+
+                // Calculate the percentage of free space
+                double freePercentage = (vgFreeGB / vgSizeGB) * 100;
+
+                LOG_TRACE("LVM Volume Group: {}, Total Size: {} GB, Free "
+                          "Space: {} GB, Free Percentage: {:.2f}%",
+                    vgName, vgSizeGB, vgFreeGB, freePercentage);
+
+                // Check if the free space is less than 50%
+                if (freePercentage < 50.0) {
+                    LOG_WARN("LVM Volume Group '{}' does not have enough free "
+                             "space (only {:.2f}% available).",
+                        vgName, freePercentage);
+                    allVGsHaveEnoughSpace = false;
+                }
+            }
+
+            if (!allVGsHaveEnoughSpace) {
+                throw std::runtime_error("LVM ERROR: Not all LVM volume groups "
+                                         "have at least 50% free space.");
+            }
+
+            LOG_INFO(
+                "LVM: All LVM volume groups have at least 50% free space.");
+        } catch (const std::exception& e) {
+            throw std::runtime_error(fmt::format(
+                "LVM ERROR: Failed to parse disk space information: {}",
+                e.what()));
+        }
+    } else {
+        throw std::runtime_error(
+            "LVM ERROR: Failed to check available disk space.");
+    }
+}
+
+void LVM::verifyAvailablePartitions()
+{
+    std::list<std::string> output;
+    const std::string checkPartitionsCommand
+        = "lsblk --noheadings --output MOUNTPOINT";
+    int exitCode = cloyster::runCommand(checkPartitionsCommand, output, false);
+
+    if (exitCode == 0) {
+        for (const auto& line : output) {
+            if (line == "/var") {
+                m_hasVarPartition = true;
+            } else if (line == "/opt") {
+                m_hasOptPartition = true;
+            } else if (line == "/home") {
+                m_hasHomePartition = true;
+            }
+        }
```

Comment:
I think a generic case should be used. We cannot foresee all the possible combinations in the world. For example: `/var/log` can also be a partition.

Reply:
Actually, I forgot to remove `/var` and `/opt`, was using only for testing. Do we need to know any other than `/home`? If not, I'm removing the other ones.

Reply:
We need to snapshot the entire volume. So that's why the partition should be populated in a vector to be analyzed.

Consider a system with `/usr` in a separate partition. If we don't snapshot it, a rollback will eventually desync from what is in other system partitions.

Usually `/home`, `/var/log` and `/tmp` can be ignored.